components/locid/src/helpers.rs

```diff
@@ -85,15 +89,16 @@ impl<T> ShortVec<T> {
             ShortVec::ZeroOne(None) => ShortVec::ZeroOne(Some(elt)),
             ShortVec::ZeroOne(Some(item)) => {
                 let items = if index == 0 {
-                    vec![elt, item]
+                    vec![elt, item].into_boxed_slice()
                 } else {
-                    vec![item, elt]
+                    vec![item, elt].into_boxed_slice()
                 };
                 ShortVec::Multi(items)
             }
-            ShortVec::Multi(mut items) => {
+            ShortVec::Multi(items) => {
+                let mut items = items.into_vec();
                 items.insert(index, elt);
-                ShortVec::Multi(items)
+                ShortVec::Multi(items.into_boxed_slice())
```

Comment:
note: I think the `realloc` call here is actually free if we're under the minimum allocation size, which seems to be 16 bytes with `malloc`. Also in `remove`.